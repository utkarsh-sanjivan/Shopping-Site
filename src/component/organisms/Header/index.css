.header {
    padding: 20px;
    background-color: #f18c6c;
    display: flex;
}

.logo-header {
    border: 1px solid #f4f4f4;
    width: 75px;
    height: 50px;
    background-image: url('./cart-logo.png');
    background-size: 100% 100%;
}

.title-header {
    width: 90%;
    height: 50px;
    line-height: 50px;
    text-align: center;
    font-size: 32px;
    color: #f7c6b5;
    font-weight: 200;
}