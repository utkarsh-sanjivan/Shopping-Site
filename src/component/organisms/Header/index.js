import React from 'react';
import './index.css';

export default function Header(props) {
    return (
        <div className='header'>
            <div className='logo-header'></div>
            <div className='title-header'>Shopping Site</div>
            <div className='cart-item-count'>
                <i className="fas fa-shopping-cart cart-icon"></i>
                {props.totalProductCount>0?<div className='total-count'>{props.totalProductCount}</div> :null}
            </div>
        </div>
    )
}