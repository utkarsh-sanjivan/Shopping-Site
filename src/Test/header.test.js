import React from 'react';
import { create } from "react-test-renderer";
import Header from '../component/organisms/Header/index';

let wrapper;
let root;
const renderComponent = props => <Header {...props} />;
const baseProps = {
    totalProductCount: 0,
};

beforeEach(() => {
  wrapper = create(renderComponent(baseProps));
  root=wrapper.root;
});

describe('Testing Header Component with 0 totalProductCount', () => {
  it('Header Component should render properly', () => {
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

    it('should atleast 4 div component', () => {
        expect(root.findAllByType('div').length===4);
    });

    it('should atleast 1 i component', () => {
        expect(root.findAllByType('i').length===1);
    });
});

describe('Testing Header Component with totalProductCount', () => {
    wrapper = create(renderComponent({ totalProductCount: 10 }));
    root=wrapper.root;
  it('Header Component should render properly', () => {
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

    it('should atleast 5 div component', () => {
        expect(root.findAllByType('div').length===5);
    });

    it('should atleast 1 i component', () => {
        expect(root.findAllByType('i').length===1);
    });
});
