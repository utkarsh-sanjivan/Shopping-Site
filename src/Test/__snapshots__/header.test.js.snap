// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Header Component Rendering Header Component should render properly 1`] = `
<div
  className="header"
>
  <div
    className="logo-header"
  />
  <div
    className="title-header"
  >
    Shopping Site
  </div>
  <div
    className="cart-item-count"
  >
    <i
      className="fas fa-shopping-cart cart-icon"
    />
    <div
      className="total-count"
    >
      10
    </div>
  </div>
</div>
`;

exports[`Testing Header Component with 0 totalProductCount Header Component should render properly 1`] = `
<div
  className="header"
>
  <div
    className="logo-header"
  />
  <div
    className="title-header"
  >
    Shopping Site
  </div>
  <div
    className="cart-item-count"
  >
    <i
      className="fas fa-shopping-cart cart-icon"
    />
  </div>
</div>
`;

exports[`Testing Header Component with totalProductCount Header Component should render properly 1`] = `
<div
  className="header"
>
  <div
    className="logo-header"
  />
  <div
    className="title-header"
  >
    Shopping Site
  </div>
  <div
    className="cart-item-count"
  >
    <i
      className="fas fa-shopping-cart cart-icon"
    />
  </div>
</div>
`;
